agent/discovery_chain_endpoint.go

```diff
@@ -0,0 +1,96 @@
+package agent
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"time"
+
+	cachetype "github.com/hashicorp/consul/agent/cache-types"
+	"github.com/hashicorp/consul/agent/structs"
+)
+
+func (s *HTTPServer) DiscoveryChainRead(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+	switch req.Method {
```

Comment:
This switch statement here should be unnecessary as the exact same check will be performed in http.go since you are setting the allowed methods to GET and POST in http_oss.go.

Reply:
Removed.